### ISSUE TITLE
[BUG] vm stuck in loop toggle between "pause"/"unpause"

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#4828]
**Describe the bug**
The VM after running for some time and having long-running automation tasks via Ansible running against it seems to be stuck in pause/unpause loop on the GUI / harvester-dashboard.
Seems rare to reproduce.

**To Reproduce**
~~Pre-reqs:~~
~~- Have the VM target a single replica non-migratable storage class~~
~~- Have the Host machine have a management network that is bonded balance-tlb~~
~~- Have the VM run on a bonded balance-tlb vm network~~
~~- Utilize an .iso based install for the VM (non-cloud-image) to disk (CD-ROM install VM to Root Disk w/ LVM)~~
~~- Run the VM with three drives:~~
  ~~- one root~~
  ~~- one mounted at some point (non-lvm)~~
  ~~- one mounted at some other point (non-lvm)~~
~~- At the switch level, assign a static IP via MAC address, ensure that the same MAC address is used on the VM~~
~~Steps:~~
~~1. Run the VM for 12+ hrs~~
~~1. Have an Ansible task / SSH task running on the VM~~
(updated):
1. Create A VM with an Extra Volume
1. Mount the Volume in the VM
1. Write data to the VM's Volume at it's mount point via SSH ( maybe something like: `dd if=/dev/random of=random.stuff bs=1000000 count=50` )
1. Keep writing data that exceeds the overall block size allocated for the Volume
1. Check the longhorn dashboard for the volume
1. The VM will enter a pause / un-pause toggle loop, with no mention to the user that the VM's mounted volume has exceeded the allocated size

**Workaround**
Stop & Restart the VM, possibly...

**Expected behavior**
That the VM did not enter a pause/unpause cycle.  Or perhaps let the user know that the Volume mounted exceeded it's allocated size - then allow for possible rescue / cleanup... ?

**Support bundle**
[supportbundle_d157486a-3253-441e-99d1-a899ad7d498d_2023-12-05T18-19-32Z.zip](https://github.com/harvester/harvester/files/13567537/supportbundle_d157486a-3253-441e-99d1-a899ad7d498d_2023-12-05T18-19-32Z.zip)

**Environment**
 - Harvester ISO version:  v1.2.1
 - Underlying Infrastructure: HP-DL160

**Comment by irishgordo** [comment]
Seeing this, bumping up "reproduce":

```
the status of pod readiness gate "kubevirt.io/virtual-machine-unpaused" is not "True", but False 
```


https://github.com/harvester/harvester/assets/5370752/eb851566-f1fa-4a3f-ac01-546f3a70b469

**Comment by irishgordo** [comment]
Perhaps, this is just a symptom of the "Actual Size" being larger than the "Size" in Longhorn on the Volume ...?

![Screenshot from 2023-12-05 16-16-47](https://github.com/harvester/harvester/assets/5370752/0dd5112e-c95a-4887-8be2-a442d9b55dcf)

**Comment by w13915984028** [comment]
A similar issue: https://github.com/harvester/harvester/issues/4633

**Comment by irishgordo** [comment]
After stopping the VM, increasing the size of the disk in Harvester to 700G.

Then hopping onto the VM:
```
$ sudo systemctl stop docker.service
$ sudo umount /dev/vda1
$ sudo sudo growpart /dev/vda 1
$ sudo e2fsck -f /dev/vda1
$ sudo resize2fs /dev/vda1
$ sudo mount /dev/vda1 /mnt/storage/docker
$ sudo systemctl start docker.service
```

And it seems, while running a:
[rancher-load-images.sh](https://github.com/rancher/rancher/releases/download/v2.6.11/rancher-load-images.sh) script in a `tmux` session

That entire vm enters a "pause" state and it's trying to "delete a snapshot in Longhorn".
The size still had headroom it seemed, 664 Used 700 Available:
```
 Size:
700 Gi
Actual Size:664 Gi
```
But  `watch -n 3 | df -h | head -n 1 && df -h | grep -ie "vd"` it seems that at least from the VM's SSH terminal to say that the "Use%" was still only 45% at /mnt/docker/storage (/dev/vda) (700G) disk.


![Screenshot from 2023-12-06 15-56-18](https://github.com/harvester/harvester/assets/5370752/2c1366d2-c651-46e7-9570-eeaf2c9bc608)



https://github.com/harvester/harvester/assets/5370752/d1c7d0b8-84f3-4bf3-84f9-cbc1709f3e9e

The "Deletion" of the snapshot finishes.
The VM gets into a stable state.

But we'v gone from consuming 45% Disk Usage on the VM on /dev/vda1 to 33%:
```
/dev/vda1                          688G  211G  443G  33% /mnt/docker/storage
```

The whole operation of the rancher-load-images.sh script seems to have not succeeded.
Disk size was increasing on /mnt/docker/storage but not on /var/lib/registry - but disk size increased on /mnt/docker/storage to 45~% but then that "seemed to have trigger" Longhorn somehow deleting the snapshot, throwing the VM into a pause/un-pause state, until the snapshot was deleted.

**Comment by irishgordo** [comment]
I think throwing more resources at it seemed to solve things - I'll close this out later if I can't reproduce, but it seems very interesting that the resources need to be much higher than that of the resources needed for something running on a single qemu/kvm hypervisor w/ Vagrant

**Comment by w13915984028** [comment]
@Vicente-Cheng @WebberHuang1118 any insights?

@irishgordo Is this reproduced later?

**Comment by w13915984028** [comment]
@irishgordo I added this issue into epic https://github.com/harvester/harvester/issues/3261, check if we need to add some enhancements.

**Comment by w13915984028** [comment]
@irishgordo I added this issue into epic https://github.com/harvester/harvester/issues/3261, check if we need to add some enhancements.

**Comment by irishgordo** [comment]
@w13915984028  I'll be able to check this on v1.3.0-head after https://github.com/harvester/dashboard/pull/944 gets merged in :smile:

**Comment by Vicente-Cheng** [comment]
Hi @irishgordo, 

It seems you are trying to write data that exceeds volume size.
If yes, I thought it meant the ENOSPC error from Longhorn.
For now, most errors will present IOError from the Longhorn side.
So your test case will get the EIO, so the VM will flap in pause and running.

UPDATE: What is the Longhorn disk size (for the volume) in your cases?

cc @w13915984028, @bk201

**Comment by w13915984028** [comment]
@Vicente-Cheng 
If we do such in a KVM VM directly, what will happen?

From https://github.com/harvester/harvester/issues/4828#issuecomment-1841865781, the LH volume size is 500G, actual size is 581 G, and the state is still `Healthy`. How can user be warned ? Does LH have kind of state to say the volume is to be full and warn user?

Meanwhile, after we further confirm this issue, need to add this to konwn issue/trouble-shooting of Harvester doc.

**Comment by Vicente-Cheng** [comment]
Hi @w13915984028, 

> If we do such in a KVM VM directly, what will happen?

No, KVM will handle the volume full situation if the Longhorn backend device is not full.

Hi @irishgordo,

According to https://github.com/harvester/harvester/issues/4828#issuecomment-1843903713, I thought your disk on the harvester node was 500G at first, then increased to 700G, right?

This issue is related to the full Longhorn backend disk.

> the LH volume size is 500G, actual size is 581 G

The size and actual size do not match because we have snapshots (no matter the user or system created).
The actual size calculates the whole size of this volume (means head + all snapshots).

And I agree that we need documents for this situation. I can add some topic for this.

**Comment by irishgordo** [comment]
@Vicente-Cheng  you're correct - originally this was seen as the size was increased manually to try to compensate for the large need of a big offline docker-registry.

On v1.3.0-rc1 with harvester/dashboard of:
```
╭─mike at suse-workstation-team-harvester in ~/Projects/dashboard on master✔
╰─± git log | head -n 5       
commit 19b69b0388eae599bc34a8e9dbd61601481ee75e
Merge: 608d04528 d9bf76d2b
Author: Francesco Torchia <francesco.torchia@suse.com>
Date:   Tue Jan 30 16:54:44 2024 +0100
```
Hooked in with (harvester/dashboard) dashboard dev mode:
```
API=https://192.168.122.70 yarn dev 
```

`dd` commands can filll the disk but will stop once it's full.
When the extra mounted volume is filled up the VM does not go into a flapping state.

It seems that the flapping state was more just encountered when the VM had one of the underlying volume's resized from 500->700 and the underlying hardware (the SATA SSD) (I believe) didn't have enough space for ( snapshots + head ).

I think with just some documents/topics to explain the situation, we could possibly just go ahead and close this out?
![Screenshot from 2024-01-30 17-23-11](https://github.com/harvester/harvester/assets/5370752/eb9a9276-5a33-458d-bb2a-782bb7ba2f6c)
![Screenshot from 2024-01-30 17-17-22](https://github.com/harvester/harvester/assets/5370752/018f8fdf-0cec-49c9-aa56-aea98685bcf4)
![Screenshot from 2024-01-30 17-16-42](https://github.com/harvester/harvester/assets/5370752/ca4e832d-9992-416d-bcf4-00788c1fb63a)
![Screenshot from 2024-01-30 17-08-56](https://github.com/harvester/harvester/assets/5370752/14d72e6d-b5a3-4529-91a4-5a852d93624a)
